undefined reference to `sys_arch_reboot'

When I build my own application, it gets error during linking to generate `zephyr_prebuilt.elf`
```
/home/hye/bin/zephyr-sdk-0.12.2/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/10.2.0/../../../../riscv64-zephyr-elf/bin/ld: libzephyr.a(reboot.c.obj): in function `sys_reboot':
/home/hye/tmp/zephyrproject/zephyr/subsys/power/reboot.c:25: undefined reference to `sys_arch_reboot'
```
but with the same prj.conf file it works in `sample/hello_world`, there are no errors. I try to find the differences of this two applications

my application:
```
/home/hye/bin/zephyr-sdk-0.12.2/riscv64-zephyr-elf/bin/riscv64-zephyr-elf-g++ CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj -o zephyr_prebuilt.elf  -Wl,-T linker.cmd -Wl,-Map=/home/hye/tmp/tvm/tests/micro/qemu/test_zephyr-host-workspace/2021-02-01T17-04-41/build/runtime/zephyr/zephyr_prebuilt.map -Wl,--whole-archive libzephyr.a ../app/libapp.a arch/common/libarch__common.a arch/arch/riscv/core/libarch__riscv__core.a lib/libc/newlib/liblib__libc__newlib.a lib/posix/liblib__posix.a subsys/random/libsubsys__random.a drivers/serial/libdrivers__serial.a -Wl,--no-whole-archive kernel/libkernel.a CMakeFiles/offsets.dir/arch/riscv/core/offsets/offsets.c.obj -L"/home/hye/bin/zephyr-sdk-0.12.2/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/10.2.0/rv32imafc/ilp32f" -L/home/hye/tmp/tvm/tests/micro/qemu/test_zephyr-host-workspace/2021-02-01T17-04-41/build/runtime/zephyr -lgcc -Wl,--print-memory-usage /home/hye/tmp/tvm/tests/micro/qemu/zephyr-runtime/libtvm__libutvm_rpc_server.a /home/hye/tmp/tvm/tests/micro/qemu/zephyr-runtime/libtvm__libutvm_rpc_common.a /home/hye/tmp/tvm/tests/micro/qemu/zephyr-runtime/libtvm__libcommon.a /home/hye/tmp/tvm/tests/micro/qemu/zephyr-runtime/libtvm__libsrc.a /home/hye/tmp/tvm/tests/micro/qemu/zephyr-runtime/libtvm__libmodule.a arch/common/libisr_tables.a -mabi=ilp32f -march=rv32imafc -Wl,--gc-sections -Wl,--build-id=none -Wl,--sort-common=descending -Wl,--sort-section=alignment -Wl,-u,_OffsetAbsSyms -Wl,-u,_ConfigAbsSyms -nostdlib -static -no-pie -Wl,-X -Wl,-N -Wl,--orphan-handling=warn -lm -Wl,-lc -L"/home/hye/bin/zephyr-sdk-0.12.2/riscv64-zephyr-elf/riscv64-zephyr-elf"/lib/rv32imafc/ilp32f -Wl,-lgcc -lc -specs=nano.specs
/home/hye/bin/zephyr-sdk-0.12.2/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/10.2.0/../../../../riscv64-zephyr-elf/bin/ld: libzephyr.a(reboot.c.obj): in function `sys_reboot':
/home/hye/tmp/zephyrproject/zephyr/subsys/power/reboot.c:25: undefined reference to `sys_arch_reboot'
```
hello world 
```
/home/hye/bin/zephyr-sdk-0.12.2/riscv64-zephyr-elf/bin/riscv64-zephyr-elf-g++ CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj -o zephyr_prebuilt.elf  -Wl,-T linker.cmd -Wl,-Map=/home/hye/tmp/zephyrproject/zephyr/samples/hello_world/build/zephyr/zephyr_prebuilt.map -Wl,--whole-archive ../app/libapp.a libzephyr.a arch/common/libarch__common.a arch/arch/riscv/core/libarch__riscv__core.a lib/libc/newlib/liblib__libc__newlib.a lib/posix/liblib__posix.a subsys/random/libsubsys__random.a drivers/serial/libdrivers__serial.a -Wl,--no-whole-archive kernel/libkernel.a CMakeFiles/offsets.dir/arch/riscv/core/offsets/offsets.c.obj -L"/home/hye/bin/zephyr-sdk-0.12.2/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/10.2.0/rv32imafc/ilp32f" -L/home/hye/tmp/zephyrproject/zephyr/samples/hello_world/build/zephyr -lgcc -Wl,--print-memory-usage arch/common/libisr_tables.a -mabi=ilp32f -march=rv32imafc -Wl,--gc-sections -Wl,--build-id=none -Wl,--sort-common=descending -Wl,--sort-section=alignment -Wl,-u,_OffsetAbsSyms -Wl,-u,_ConfigAbsSyms -nostdlib -static -no-pie -Wl,-X -Wl,-N -Wl,--orphan-handling=warn -lm -Wl,-lc -L"/home/hye/bin/zephyr-sdk-0.12.2/riscv64-zephyr-elf/riscv64-zephyr-elf"/lib/rv32imafc/ilp32f -Wl,-lgcc -lc -specs=nano.specs
```
still can't solve it, can anybody help me, thanks!

--- a/soc/riscv/riscv-privilege/virt/CMakeLists.txt
+++ b/soc/riscv/riscv-privilege/virt/CMakeLists.txt
@@ -1,3 +1,3 @@
 # SPDX-License-Identifier: Apache-2.0
 
-zephyr_sources()
+zephyr_sources(soc.c)

--- a/soc/riscv/riscv-privilege/virt/soc.c
+++ b/soc/riscv/riscv-privilege/virt/soc.c
@@ -1,0 +1,36 @@
+/*
+ * Copyright (c) 2021 Katsuhiro Suzuki
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief QEMU RISC-V virt machine hardware depended interface
+ */
+
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <sys/util.h>
+
+/*
+ * Exit QEMU and tell error number.
+ *   Higher 16bits: indicates error number.
+ *   Lower 16bits : set FINISHER_FAIL
+ */
+#define FINISHER_FAIL		0x3333
+
+/* Exit QEMU successfully */
+#define FINISHER_EXIT		0x5555
+
+/* Reboot machine */
+#define FINISHER_REBOOT		0x7777
+
+void sys_arch_reboot(int type)
+{
+	volatile uint32_t *reg = (uint32_t *)SIFIVE_SYSCON_TEST;
+
+	*reg = FINISHER_REBOOT;
+
+	ARG_UNUSED(type);
+}

--- a/soc/riscv/riscv-privilege/virt/soc.h
+++ b/soc/riscv/riscv-privilege/virt/soc.h
@@ -10,6 +10,7 @@
 #include <soc_common.h>
 #include <devicetree.h>
 
+#define SIFIVE_SYSCON_TEST           0x00100000
 #define RISCV_MTIME_BASE             0x0200BFF8
 #define RISCV_MTIMECMP_BASE          0x02004000
 

